src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1093,6 +1094,48 @@ class spell_warr_sword_and_board : public SpellScriptLoader
         }
 };
 
+// 215538 - Trauma
+class spell_warr_trauma : public SpellScriptLoader
+{
+public:
+    spell_warr_trauma() : SpellScriptLoader("spell_warr_trauma") { }
+
+    class spell_warr_trauma_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_warr_trauma_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_TRAUMA_EFFECT
+            });
+        }
+
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            if (Unit* target = eventInfo.GetActionTarget())
+            {
+                //Get the Remaining Damage from the aura (if exist)
+                int32 remainingDamage = target->GetRemainingPeriodicAmount(target->GetGUID(), SPELL_WARRIOR_TRAUMA_EFFECT, SPELL_AURA_PERIODIC_DAMAGE);
+                //Get 25% of damage from the spell casted (Slam & Whirlwind) plus Remaining Damage from Aura
+                int32 damage = CalculatePct(eventInfo.GetDamageInfo()->GetDamage(), aurEff->GetAmount()) + remainingDamage;
```

Comment:
The calculated amount is wrong here, value you pass to `CastCustomSpell` is PER TICK amount so you are making it deal 25% damage of Slam on each tick (`GetRemainingPeriodicAmount` returns the value already divided so just adding it to pre-tick value is correct)